packages/ui-library/src/components/stepper/stories.js

```diff
@@ -46,20 +66,112 @@ export const Factory = {
 	},
 };
 
-export default {
-	title: "2) Components/Stepper",
-	component: Stepper,
-	argTypes: {
-		className: { control: "text" },
-		currentStep: { control: "number" },
+export const StepsProp = {
+	parameters: {
+		controls: { disable: false },
+	},
+	render: ( { className } ) => {
+		const [ isComplete, setIsComplete ] = useState( false );
+		const [ currentStep, setCurrentStep ] = useState( 0 );
+		const [ stepsList, setStepsList ] = useState( [
+			{ children: "INSTALL", isComplete: false, isActive: true },
+			{ children: "ACTIVATE", isComplete: false, isActive: false },
+			{ children: "SET UP", isComplete: false, isActive: false },
+			{ children: "CONNECT", isComplete: false, isActive: false },
+		] );
+		const handleNext = useCallback( () => {
+			if ( currentStep < stepsList.length - 1 ) {
+				const tempSteps = [ ...stepsList ];
+				tempSteps[ currentStep ].isComplete = true;
+				tempSteps[ currentStep ].isActive = false;
+				tempSteps[ currentStep + 1 ].isActive = true;
+				setIsComplete( false );
+				setCurrentStep( currentStep + 1 );
+				setStepsList( tempSteps );
+			} else if ( currentStep === stepsList.length - 1 && ! isComplete ) {
+				const tempSteps = [ ...stepsList ];
+				tempSteps[ currentStep ].isComplete = true;
+				tempSteps[ currentStep ].isActive = false;
+				setIsComplete( true );
+				setStepsList( tempSteps );
+			} else if ( isComplete ) {
+				const tempSteps = stepsList.map( ( step ) => ( {
+					...step,
+					isComplete: false,
+					isActive: false,
+				} ) );
+				tempSteps[ 0 ].isActive = true;
+				setIsComplete( false );
+				setCurrentStep( 0 );
+				setStepsList( tempSteps );
+			}
+		}, [ setIsComplete, setStepsList, setCurrentStep, currentStep, stepsList, isComplete ] );
```

Comment:
😅 nice how the implementation can highlight some potential API friction.

`isComplete` can be calculated by `currentStep >= stepsList.length - 1`, then you don't need to have it as global state. Or if using individual steps, you might have to loop over all individual `isComplete`.

`isActive` per step is never really possible? So should that not be `currentStep` in the context and then have its own index to verify against? I suppose this is a clearer picture, but it might also lead to errors where you can just activate multiple steps at the same time.